### PR TITLE
[MOB-41] - POST Register endpoint

### PR DESCRIPTION
**Pull request by JM-Abrodos**
## Pull Request Checklist

- [x] **Title**
- [x] **[MOB US' Link](https://mobilemakers.atlassian.net/browse/MOB-41)**
- [ ] **Link PR to US**
- [x] **Add Tests**
- [ ] **Add ENV Variables and Validations**
- [ ] **Check Story Points**
- [ ] **Documentation Updates**
- [x] **Swagger Updates**

## Testing Guide
Install new dependencies
```
npm i
```

Example curl:
```
curl --location 'http://localhost:3000/v1/account/register' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "John",
    "lastName": "Donnel-Vic",
    "password": "SuperSecret1!",
    "email": "ozounalocal5@yopmail.com"
}'
```

### CODE REVIEW

File: src/infrastructure/http/controllers/account.controller.ts [hunk 1]
```diff
@@ -37,12 +40,53 @@ export class AccountController {
...
     @Body() accountLoginDto: AccountLoginDto,
+    @RequestId() requestId: string,
   ): Promise<AccountLoggedDto> {
     try {
-      const response = await this.loginService.login(accountLoginDto);
+      this.logger.info(
+        `[AccountController] [login] - x-request-id:${requestId}`,
+      );
+
+      const response: AccountLoggedDto = await this.loginService.login(
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/account.controller.ts:50]: The function is already typed, it is not necessary to add it in the answer. Same with register.

File: src/infrastructure/http/dtos/account-register.dto.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+export class AccountRegisterDto {
+  @Matches(ACCOUNT_REGISTER_REGEX.NAME, {
+    message: `name only accept letters (Uppercase or lowercase), spaces and ''',  '-'`,
+  })
+  @MinLength(3, { message: 'name must have min 3 characters' })
+  @MaxLength(100, { message: 'name must have max 100 characters' })
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'John' })
```
Comment (LuciaAccenture) [inline src/infrastructure/http/dtos/account-register.dto.ts:26]: Remember to add the max and min length in all the @Apiproperty

File: src/infrastructure/http/swagger/headers/account-headers.ts [hunk 3]
```diff
@@ -3,3 +3,7 @@ import { CONTENT_TYPE } from './common';
 export const LOGIN_HEADERS = {
   ...CONTENT_TYPE,
 };
+
+export const REGISTER_HEADER = {
+  ...CONTENT_TYPE,
+};
```
Comment (LuciaAccenture) [inline src/infrastructure/http/swagger/headers/account-headers.ts:9]: If we are going to put all headers that have the same path v1/xxxxxx in the same file, we should add the refresh here and do the same at the events file.

File: test/mocks/test.env.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export default {
+  MONGODB_URI: 'mongodb://localhost:27017/esmorga',
+  APP_PORT: 3000,
+  NODE_ENV: 'TEST',
+  JWT_SECRET:
+    'j1u5RDS8Ga4hzDcS1vGULNPHYMGjMfjINLarWJM9UIjAVPxgFUgA1tMc/OT4NItCzGqU/cGlkjg4l6kgqcNEcQ',
+  ACCESS_TOKEN_TTL: 600,
+  MAX_PAIR_OF_TOKEN: 5,
+  SWAGGER_JSON_SCHEMA_URL: 'https://qa.esmorga.canarte.org/swagger-json',
```
Comment (LuciaAccenture) [inline test/mocks/test.env.ts:9]: This variable doesn't exist in the code.

File: src/domain/services/encode-value.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { createHash } from 'crypto';
+
+export function encodeValue(value: string) {
+  return createHash('sha256').update(value).digest('hex');
+}
```
[the remark this answers is not in the document]
Comment (JM-Abrodos) [reply src/domain/services/encode-value.ts:5]: It's true we are only hashing the password right now and probably we will not hash other field but we will overwrite this logic in this ticket: https://mobilemakers.atlassian.net/browse/MOB-75

**Review by LuciaAccenture — APPROVED** [review]

**Pull request merged**